src/types.ts

```diff
@@ -61,13 +61,30 @@ export interface Driver<OptionsT = any, InstanceT = any> {
   watch?: (callback: WatchCallback) => MaybePromise<Unwatch>;
 }
 
+type NsStorageValue = {
+  data: unknown;
+  [key: string]: unknown;
+};
+
+type NsDataType<T extends NsStorageValue> = T["data"];
```

Comment:
I'm not super-excited about the `Ns` prefix. Really `StorageMeta` would have been perfect but it's already used. So maybe something like `StorageDefinition` and `StorageItemMap`?

Reply:
What about TypedStorage?

Reply:
That could work! But in a sense `Typed` is true by definition as it's a type. So _what_ is being typed? I guess it partly depends on what else is in there, but `Definition` suggests that it totally describes the storage setup (which is why you wanted `data` as a key within it, right?)....